test/node-rpc-test.js

```diff
@@ -0,0 +1,230 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+'use strict';
+
+const assert = require('bsert');
+const FullNode = require('../lib/node/fullnode');
+const {NodeClient, WalletClient} = require('hs-client');
+const KeyRing = require('../lib/primitives/keyring');
+const pkg = require('../lib/pkg');
+
+// will this work?
+const blake2b = require('bcrypto/lib/blake2b');
+const base64 = require('bcrypto/lib/encoding/base64');
+
+const MAGIC_STRING = `${pkg.currency} signed message:\n`;
+
+const ports = {
+  p2p: 49331,
+  node: 49332,
+  wallet: 49333
+};
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  workersSize: 2,
+  plugins: [require('../lib/wallet/plugin')],
```

Comment:
Including a wallet here seems excessive to me, since it's just used to derive a coinbase address, which could be hard-coded and reduce test complexity, but it doesn't bother me that much.